### DISCUSSION TITLE
Problem with running MCP demos

### DISCUSSION THREAD

**Discussion by PrionMike** [lindeloev/mcp#161] [General]
I love the concepts embodied in mcp, and have some specific use cases in mind. When trying to run the demos though, I ran into a puzzling issue. When I call `fit = mcp(model, data = ex$data)` as per code cut and pasted from the first Brief Worked Example in the tutorial at https://github.com/lindeloev/mcp I get the error message 

> Error in func_eval(data[, column], na.rm = TRUE) : could not find function "func_eval"

The internal call `func_eval(data[, column], na.rm = TRUE)` can be found in mcp function `R/run_jags.R` (see https://rdrr.io/github/lindeloev/mcp/src/R/run_jags.R), in the code block headed with the comment 

```r
# Add e.g. MINX = min(data$x) for all variables where they are used.
# Searches whether it is in jags_code. If yes, add to jags_data
# TO DO: there must be a prettier way to do this.
```

I am running mcp 0.3.3 on R 4.2.3 in RStudio "Spotted Wakerobin" Release (e7373ef8, 2022-09-06) for Windows, on Windows 10. Thanks for any suggestions!

**Comment by PrionMike** [comment]
I probably should have mentioned that I can run the code in the Coal Mining Disasters vignette (_https://lindeloev.github.io/mcp/articles/poisson.html_) perfectly.

**Comment by lindeloev** [comment]
Thanks for reporting! Perhaps you defined the variable `sd`, `min`, `max`, or `mean`, overwriting the built-in function? I can reproduce your error if I define `sd = 5` before calling `mcp(...)`.

**Comment by lindeloev** [comment]
If that's not the issue, let's see if we can pinpoint this further. If you open a blank session in R 4.3.1 and run the following in RStudio with mcp 0.3.3, do you get the error? I just installed mcp 0.3.3 in a "fresh" R 4.3.1 on RStudio 2023.06 on Windows 11 and it ran as expected.

```r
library(mcp)

# Define the model
model = list(
  response ~ 1,  # plateau (int_1)
  ~ 0 + time,    # joined slope (time_2) at cp_1
  ~ 1 + time     # disjoined slope (int_3, time_3) at cp_2
)

# Get example data and fit it
ex = mcp_example("demo")
fit = mcp(model, data = ex$data)
```

**Comment by PrionMike** [comment]
Thank you so much for getting back to me Jonas, and there is absolutely no
cause for apologies! :)

Although I have not yet pinpointed exactly what was wrong that caused my
code to fail, your recommendation to start a blank session showed that
there must have been something in my (rather overloaded) saved RStudio
workspace that was causing a conflict of the type you suggested. When I ran
the code you supplied in that blank session it worked perfectly. Then, when
I re-loaded my own workspace and cleared the environment of all
nonessential variables, it worked perfectly there too.

It seems the good news is that there is no bug in mcp -- only a local
conflict that I created inadvertently myself! I will make a note of this on
GitHub, so it hopefully does not cause any further confusion or waste more
of your time.

Thank you so much, and again, I love the work you have done with mcp!

All the best,

mike
